src/platform/site-wide/user-nav/components/SearchMenu.jsx

```diff
@@ -49,89 +54,133 @@ export class SearchMenu extends React.Component {
     }
   }
 
-  async getSuggestions() {
+  getSuggestions = async () => {
     const { userInput } = this.state;
 
+    // end early / clear suggestions if user input is too short
     if (userInput?.length <= 2) {
       if (this.state.suggestions.length > 0) {
         this.setState({ suggestions: [] });
       }
 
       return;
     }
+
+    // encode user input for query to suggestions url
     const encodedInput = encodeURIComponent(userInput);
+
+    // fetch suggestions
     try {
       const response = await fetch(
         `${searchGovSuggestionEndpoint}?=&name=va&q=${encodedInput}`,
       );
 
       const suggestions = await response.json();
       this.setState({ suggestions });
+
+      // if we fail to fetch suggestions
     } catch (error) {
       Sentry.captureException(error);
     }
-  }
+  };
 
   handleInputChange = event => {
     this.setState({ userInput: event.target.value });
   };
 
-  handleKeyUp = event => {
-    if (
-      (event.which || event.keyCode) === ENTER_KEY &&
-      document.getElementById('query') === document.activeElement
-    ) {
-      this.handleSearchEvent();
+  // function to control state changes within the downshift component
+  handelDownshiftStateChange = (changes, state) => {
+    // when a user presses the escape key, clear suggestions and close the dropdown
+    if (changes.type === Downshift.stateChangeTypes.keyDownEscape) {
+      this.setState({
+        highlightedIndex: null,
+        suggestions: [],
+      });
       return;
     }
-    if ((event.which || event.keyCode) === ESCAPE_KEY) {
-      this.props.clickHandler();
+    // when a user presses the enter key, exit early to prevent out highlighted suggestion being cleared
+    if (changes.type === Downshift.stateChangeTypes.keyDownEnter) {
+      return;
+    }
+
+    // update our highlighted suggestion for search logging
+    this.setState({ highlightedIndex: state.highlightedIndex });
+  };
+
+  // function to handle searches kicked off by key presses
+  handleKeyUp = event => {
+    const { highlightedIndex, suggestions } = this.state;
+
+    // kick off a search when an enter key is pressed
+    if ((event.which || event.keyCode) === ENTER_KEY) {
+      // if our highlighted index is null, search using the userInput (pass in undefined)
+      // otherwise pass in our chosen suggestion
+      const query =
+        highlightedIndex !== null ? suggestions[highlightedIndex] : undefined;
+      this.handleSearchEvent(query);
     }
   };
```

Comment:
I am probably oversimplifying things but I feel like we are overlooking something with DownShift. It isn't a _great_ component but I thought we would get at least the escape-key handling for free. Then the ENTER key we would get for free if it were in a `<form>` element.

```jsx
// Standard default submit (no calling event.preventDefault)
<form action="https://www.va.gov/search" onSubmit={captureAnalytics} ref={formRef}>
  // onChange => loadSuggestions
  // onSelect => set the input to the value of the clicked suggestion. Then just formRef.submit()
  <DownShift>...</DownShift>
</form>
```

Reply:
I have been experimenting with some of this, but hadn't tried wrapping downshift in a form element. As this PR needs to get merged ASAP (we are trying to get this POC launched Monday) I don't think I am going to be able to try these out more at this time. Next week I will see if I can't give these a try and still meet the complex acceptance criteria that 508 has laid out for us / still accomplish the analytics tagging requirements.